lib/counter_culture/reconciler.rb

```diff
@@ -257,11 +257,11 @@ def join_clauses(where)
         # store joins in an array so that we can later apply column-specific
         # conditions
         join_clauses = reverse_relation.each_with_index.map do |cur_relation, index|
-          reflect = relation_reflect(cur_relation)
+          reflect, model = relation_reflect_and_model(cur_relation)
 
-          target_table = quote_table_name(reflect.active_record.table_name)
+          target_table = quote_table_name(model.table_name)
           target_table_alias = parameterize(target_table)
-          if relation_class.table_name == reflect.active_record.table_name
+          if relation_class.table_name == model.table_name
```

Comment:
Am I reading this right that these `reflect.active_record` -> `model` changes don't change behavior, they just take the short-cut straight to `model` now that we have it here?

Reply:
Makes sense, thank you.